cmd/kubeadm/app/cmd/token.go

```diff
@@ -111,20 +103,24 @@ func NewCmdToken(out io.Writer, errW io.Writer) *cobra.Command {
 		Run: func(tokenCmd *cobra.Command, args []string) {
 			if len(args) > 0 {
 				bto.TokenStr = args[0]
+			} else {
+				var err error
+				bto.TokenStr, err = bootstraputil.GenerateBootstrapToken()
+				kubeadmutil.CheckErr(err)
 			}
 			klog.V(1).Infoln("[token] validating mixed arguments")
 			err := validation.ValidateMixedArguments(tokenCmd.Flags())
 			kubeadmutil.CheckErr(err)
 
-			err = bto.ApplyTo(cfg)
+			bootstrapToken, err := kubeadmapi.NewBootstrapTokenString(bto.TokenStr)
 			kubeadmutil.CheckErr(err)
 
 			klog.V(1).Infoln("[token] getting Clientsets from kubeconfig file")
 			kubeConfigFile = cmdutil.GetKubeConfigPath(kubeConfigFile)
 			client, err := getClientset(kubeConfigFile, dryRun)
 			kubeadmutil.CheckErr(err)
 
-			err = RunCreateToken(out, client, cfgPath, cfg, printJoinCommand, kubeConfigFile)
+			err = RunCreateToken(out, client, cfgPath, bootstrapToken, printJoinCommand, kubeConfigFile)
```

Comment:
`cfgPath` is unused here. We need to check if `cfgPath` is not empty and load an InitConfiguration and use the tokens from there if it isn't.